append nil to a slice

The following program `sample.go` triggers a panic:

```go
package main

import "fmt"

func main() {
    a := []*int{}
    a = append(a, nil)
    fmt.Println(a)
}
```

Expected result:
```console
$ go run ./sample.go
[<nil>]
```

Got:
```console
$ yaegi ./sample.go
/Users/marc/sample.go:7:6: panic
run: reflect.Value.Convert: value of type interface {} cannot be converted to type *int
goroutine 1 [running]:
runtime/debug.Stack(0x1, 0x14000090400, 0x40)
	runtime/debug/stack.go:24 +0x88
github.com/traefik/yaegi/interp.(*Interpreter).eval.func1(0x14000099bf0)
	github.com/traefik/yaegi/interp/interp.go:503 +0x94
panic(0x102e79ec0, 0x14000116770)
	runtime/panic.go:965 +0x14c
github.com/traefik/yaegi/interp.runCfg.func1(0x140003720b0, 0x14000122e00, 0x140000999c0)
	github.com/traefik/yaegi/interp/run.go:185 +0x1e8
panic(0x102e79ec0, 0x14000116770)
	runtime/panic.go:965 +0x14c
reflect.Value.Convert(0x102ea30a0, 0x14000116730, 0x194, 0x102fde708, 0x102e5c9e0, 0x102c9889e, 0x140000998c8, 0x1027a7d04)
	reflect/value.go:2463 +0x1a4
github.com/traefik/yaegi/interp.genValueAs.func1(0x140003720b0, 0x14000112640, 0x140001245a0, 0x197)
	github.com/traefik/yaegi/interp/value.go:167 +0x6c
github.com/traefik/yaegi/interp._append.func2(0x140003720b0, 0x14000436a80)
	github.com/traefik/yaegi/interp/run.go:2974 +0xa0
github.com/traefik/yaegi/interp.runCfg(0x14000122e00, 0x140003720b0)
	github.com/traefik/yaegi/interp/run.go:191 +0x8c
github.com/traefik/yaegi/interp.(*Interpreter).run(0x14000370000, 0x14000122600, 0x14000372000)
	github.com/traefik/yaegi/interp/run.go:122 +0x244
github.com/traefik/yaegi/interp.(*Interpreter).eval(0x14000370000, 0x140001e0200, 0x76, 0x16d9b37ce, 0x1b, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	github.com/traefik/yaegi/interp/interp.go:595 +0x5e8
github.com/traefik/yaegi/interp.(*Interpreter).EvalPath(0x14000370000, 0x16d9b37ce, 0x1b, 0x140001e0100, 0x76, 0x0, 0x0, 0x0)
	github.com/traefik/yaegi/interp/interp.go:417 +0xc4
main.runFile(0x14000370000, 0x16d9b37ce, 0x1b, 0x0, 0x0, 0x0)
	github.com/traefik/yaegi/cmd/yaegi/run.go:139 +0x84
main.run(0x140001c2010, 0x1, 0x1, 0x14000182480, 0x14000073ea8)
	github.com/traefik/yaegi/cmd/yaegi/run.go:102 +0x810
main.main()
	github.com/traefik/yaegi/cmd/yaegi/yaegi.go:144 +0x324
```

--- a/_test/append4.go
+++ b/_test/append4.go
@@ -1,0 +1,12 @@
+package main
+
+import "fmt"
+
+func main() {
+	a := []*int{}
+	a = append(a, nil)
+	fmt.Println(a)
+}
+
+// Output:
+// [<nil>]

--- a/interp/type.go
+++ b/interp/type.go
@@ -1731,6 +1731,9 @@ func isSendChan(t *itype) bool {
 }
 
 func isArray(t *itype) bool {
+	if t.cat == nilT {
+		return false
+	}
 	k := t.TypeOf().Kind()
 	return k == reflect.Array || k == reflect.Slice
 }

--- a/interp/value.go
+++ b/interp/value.go
@@ -162,9 +162,17 @@ func genValueAsFunctionWrapper(n *node) func(*frame) reflect.Value {
 }
 
 func genValueAs(n *node, t reflect.Type) func(*frame) reflect.Value {
-	v := genValue(n)
+	value := genValue(n)
+
 	return func(f *frame) reflect.Value {
-		return v(f).Convert(t)
+		v := value(f)
+		switch v.Type().Kind() {
+		case reflect.Chan, reflect.Func, reflect.Interface, reflect.Ptr, reflect.Map, reflect.Slice:
+			if v.IsNil() {
+				return reflect.New(t).Elem()
+			}
+		}
+		return v.Convert(t)
 	}
 }
 

--- a/interp/value.go
+++ b/interp/value.go
@@ -167,7 +167,7 @@ func genValueAs(n *node, t reflect.Type) func(*frame) reflect.Value {
 	return func(f *frame) reflect.Value {
 		v := value(f)
 		switch v.Type().Kind() {
-		case reflect.Chan, reflect.Func, reflect.Interface, reflect.Ptr, reflect.Map, reflect.Slice:
+		case reflect.Chan, reflect.Func, reflect.Interface, reflect.Ptr, reflect.Map, reflect.Slice, reflect.UnsafePointer:
 			if v.IsNil() {
 				return reflect.New(t).Elem()
 			}

